server/services/star.js

```diff
@@ -255,7 +255,11 @@ module.exports = class StarService extends EventEmitter {
         // If both stars have warp gates and they are both owned by players...
         if (sourceStar.warpGate && destinationStar.warpGate && sourceStar.ownedByPlayerId && destinationStar.ownedByPlayerId) {
             // If both stars are owned by the player then carriers can always move at warp.
-            if (destinationStar.ownedByPlayerId.equals(player._id) && sourceStar.ownedByPlayerId.equals(player._id)) {
+            let sourceAllied = getDiplomaticStatusToPlayer(game, sourceStar.ownedByPlayerId, carrier.ownedByPlayerId) && this.diplomacyService.isFormalAlliancesEnabled(game) || sourceStar.ownedByPlayerId === carrier.ownedByPlayer
```

Comment:
`getDiplomaticStatusToPlayer` returns an object. You need to check for `actualStatus`.

Maybe it'll be better to call the `isDiplomaticStatusToPlayersAllied` function as that returns a boolean.

Reply:
Put the `sourceStar.ownedByPlayerId === carrier.ownedByPlayer` bit first.

Also it should compare using the `.equals` function as we do in other places.

Reply:
Are you missing brackets?

`owned by player || (is alliances enabled && diplomatic status is allied)`?

Reply:
Not really due to how it works

A && B || C
This is the same as (A && B) || C